extensions-core/druid-bloom-filter/src/main/java/org/apache/druid/query/aggregation/bloom/BloomFilterAggregatorFactory.java

```diff
@@ -0,0 +1,256 @@
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.primitives.Ints;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.druid.guice.BloomFilterSerializersModule;
+import org.apache.druid.java.util.common.RE;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.query.ColumnSelectorPlus;
+import org.apache.druid.query.aggregation.Aggregator;
+import org.apache.druid.query.aggregation.AggregatorFactory;
+import org.apache.druid.query.aggregation.AggregatorUtil;
+import org.apache.druid.query.aggregation.BufferAggregator;
+import org.apache.druid.query.aggregation.bloom.types.BloomFilterAggregatorColumnSelectorStrategy;
+import org.apache.druid.query.aggregation.bloom.types.BloomFilterAggregatorColumnSelectorStrategyFactory;
+import org.apache.druid.query.cache.CacheKeyBuilder;
+import org.apache.druid.query.dimension.DimensionSpec;
+import org.apache.druid.query.filter.BloomKFilter;
+import org.apache.druid.segment.ColumnSelectorFactory;
+import org.apache.druid.segment.DimensionHandlerUtils;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+public class BloomFilterAggregatorFactory extends AggregatorFactory
+{
+  private static final int DEFAULT_NUM_ENTRIES = 1500;
+  protected static final BloomFilterAggregatorColumnSelectorStrategyFactory STRATEGY_FACTORY =
+      new BloomFilterAggregatorColumnSelectorStrategyFactory();
+
+  private final String name;
+  private final DimensionSpec field;
+  private final int maxNumEntries;
+
+  @JsonCreator
+  public BloomFilterAggregatorFactory(
+      @JsonProperty("name") String name,
+      @JsonProperty("field") final DimensionSpec field,
+      @Nullable @JsonProperty("maxNumEntries") Integer maxNumEntries
+  )
+  {
+    this.name = name;
+    this.field = field;
+    this.maxNumEntries = maxNumEntries != null ? maxNumEntries : DEFAULT_NUM_ENTRIES;
+  }
+
+  @Override
+  public Aggregator factorize(ColumnSelectorFactory columnFactory)
+  {
+    ColumnSelectorPlus<BloomFilterAggregatorColumnSelectorStrategy> selectorPlus =
+        DimensionHandlerUtils.createColumnSelectorPlus(
+            STRATEGY_FACTORY,
+            field,
+            columnFactory
+        );
+
+    return new BloomFilterAggregator(selectorPlus, maxNumEntries);
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(ColumnSelectorFactory columnFactory)
+  {
+    ColumnSelectorPlus<BloomFilterAggregatorColumnSelectorStrategy> selectorPlus =
+        DimensionHandlerUtils.createColumnSelectorPlus(
+            STRATEGY_FACTORY,
+            field,
+            columnFactory
+        );
+
+    return new BloomFilterBufferAggregator(selectorPlus, maxNumEntries);
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    return (Comparator<Object>) (o1, o2) -> {
```

Comment:
this comparator looks fairly expensive, let's mention that in the docs

Reply:
Added docs, also improved the set bit counting so it shouldn't be quite as expensive as it was previously.

Reply:
Wrapped with `Comparators.nullsFirst`